src/RadhydroSimulation.hpp

```diff
@@ -514,61 +519,325 @@ void RadhydroSimulation<problem_t>::advanceHydroAtLevel(int lev, amrex::Real tim
 		fluxScaleFactor = 1.0;
 	}
 
-	// update ghost zones [old timestep]
-	fillBoundaryConditions(state_old_[lev], state_old_[lev], lev, time);
+  std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
+  amrex::MFIter::allowMultipleMFIters(true);
+
+  #pragma omp parallel sections
+  {
+    #pragma omp section
+    {
+      AMREX_HIP_OR_CUDA(AMREX_HIP_SAFE_CALL (hipSetDevice(device_id));,
+                        AMREX_CUDA_SAFE_CALL(cudaSetDevice(device_id)); );
+
+      //  update ghost zones [old timestep]
+      fillBoundaryConditions(state_old_[lev], state_old_[lev], lev, time);
+      // check state validity
+      AMREX_ASSERT(!state_old_[lev].contains_nan(0, state_old_[lev].nComp()));
+      AMREX_ASSERT(!state_old_[lev].contains_nan()); // check ghost cells
+
+      for (amrex::MFIter iter(state_new_[lev]); iter.isValid(); ++iter) {
+        const amrex::Box &realZones = iter.validbox(); // 'validbox' == exclude ghost zones
+
+          //6 should be 2*ndims
+        for (int i=1; i<=AMREX_SPACEDIM*2; ++i) {
+          amrex::Box computeRange = realZones;
+
+          computeRange.grow(nghost_);
```

Comment:
I think this line is the source of all the test failures. If you print out each outer box with `amrex::Print() << computeRange << std::endl;` on a 1D test problem, you should see that this screws up all the indices of the outer boxes.

Reply:
On my fork, I just deleted this line, and almost everything works. There are still test failures for problems that have blocking factors that are less than 2*nghosts_, which is an edge case that this code ignores.